### ISSUE TITLE
DLL issues with Octave 4.2.2 on Windows 7 (PTB 3.0.14)

### ISSUE THREAD

**Issue by alexsbaldwin** [Psychtoolbox-3/Psychtoolbox-3#504]
I usually use PTB with Matlab, but I have been helping a collaborator get one of my experiments set up on an Octave system in his lab. We ran into a peculiar .dll issue that I thought worth sharing. On his computer the Screen .mex file was failing because it could not find g_dir_open in libglib-2.0-0.dll ("Entry Point Not Found"). I installed Octave 4.2.2 on my machine as well to match him and had the same problem. I took a look at Screen.mex with Dependency Walker. There were warnings, but nothing relating to the error raised by Octave when you tried to call Screen. I then took a look at every version of libglib-2.0-0.dll on my machine. There were four in total: one from gstreamer, one from GtkSharp, one from GTK2, and one in the Octave-4.2.2 folder. Curiously, the dll in the octave folder _was_ in fact missing the g_dir_open function, it had only g_dir_open_utf8. I renamed the Octave dll and tried Screen again, at which point I got a similar error: g_module_open was not found in libgmodule-2.0-0.dll. This turned out to be a similar issue, g_module_open_utf8 was there, but g_module_open was not. The dll files included in GStreamer have both functions.

The bottom line is: Octave 4.2.2 seems to come with libglib and libgmodule dlls that will prevent Psychtoolbox working if Screen.mex chooses to access them rather than the dlls from GStreamer (by the looks of things the GTKSharp and GTK2 dlls would also have been fine). My collaborator had installed Octave and Psychtoolbox on a fresh new laptop, so it may be that those incomplete dlls would be accessed by default. This is not an area that I have much experience in, but it took me a little while to get to the bottom of it, so I thought it worth sharing here in case anyone else runs into this (or if something needs to be done about it).

**Comment by kleinerm** [comment]
Hi,

thanks for trying to be helpful, but Octave 4.2.2 is not supported. The requirements page says Octave 4.2.0 and links to it. Unfortunately there isn't much stability in the dll's etc. shipped by Octave on Windows, nor in how mex files are loaded and find the dll's, so atm. the mex files i build are locked to exactly the stated version. I intend to support the latest Octave 4.4.x for Windows with the next PTB release, but again it will be locked to the specific .x version.
